pynestml/symbols/unit_type_symbol.py

```diff
@@ -156,14 +154,16 @@ def get_conversion_factor(cls, to, _from):
         factor = (_from / to).si.scale
         return factor
 
-    def is_castable_to(self, other_type):
+    def is_castable_to(self, _other_type):
+        if super(UnitTypeSymbol, self).is_castable_to(_other_type):
+            return True
         from pynestml.symbols.real_type_symbol import RealTypeSymbol
-        if other_type.is_instance_of(RealTypeSymbol):
+        if _other_type.is_instance_of(RealTypeSymbol):
```

Comment:
again, this is very surprising to me... a real should not be castable to anything but a real